Zend/zend_generators.c

```diff
@@ -824,7 +814,20 @@ ZEND_API void zend_generator_resume(zend_generator *orig_generator) /* {{{ */
 		zend_generator_restore_call_stack(generator);
 	}
 
-	/* Resume execution */
+	/* Resume execution
+	 *
+	 * Increment opline to point after the CREATE_GENERATOR / YIELD / YIELD_FROM
+	 * op, unless an exception was thrown (opline is EG(exception_op)), or the
+	 * generator is being force-closed (opline points to the start of a finally
+	 * block).
+	 */
+	if (EXPECTED(generator->execute_data->opline->opcode != ZEND_HANDLE_EXCEPTION
```

Comment:
The check for ZEND_HANDLE_EXCEPTION is not really necessary as EG(exception_op) has a couple ops at once. Only thing needing check is the force close.

Though, for that case, you can just decrement the opline in zend_generator_dtor_storage before calling this function, eliminating the need for the branch altogether.

Reply:
Nice!